### PR TITLE
WIP: Chores/detox

### PR DESCRIPTION
**Pull request by snack-0verflow**
# Description

Getting Detox setup to run e2e test in CI.

To run this locally:
(Make sure you have `detox-cli` installed, `npm i -g detox-cli`)
1. `detox build`
2. `yarn test:e2e:ios`

TODO:
- [ ] Fix Broken E2E test
- [ ] Build iOS app & run e2e tests
- [ ] Setup Android configuration
- [ ] Build Android app & run e2e tests

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: e2e/environment.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { DetoxCircusEnvironment, SpecReporter, WorkerAssignReporter } = require('detox/runners/jest-circus');
+
+class CustomDetoxEnvironment extends DetoxCircusEnvironment {
+  constructor(config, context) {
+    super(config, context);
+
+    // Can be safely removed, if you are content with the default value (=300000ms)
+    this.initTimeout = 300000;
```
Comment (bionara) [inline e2e/environment.js:8]: set 300000 as a const to explain and remove comment

**Review by bionara — DISMISSED** [review]
Nice one, looks like a good tool to have for sure!

File: e2e/environment.js (hunk 1 above)
Comment (snack-0verflow) [reply e2e/environment.js:8]: Nice one! Will make the change

**Comment by bionara** [comment]
Updated from master again. Keeping this up to date and conflict-free until we make time to address it and pull it back in :)

**Comment by bionara** [comment]
I will sort this next week.note to self/all.

**Comment by bionara** [comment]
Got this working locally, and wrote a bunch of basic tests off the landing page:
image

I still need to figure out how to make a proper binaryPath for the detox config, in order that other devs can run locally: ATM the one committed doesn't work for me and I need to use a hardcoded path from xcode - will get to the bottom of it.
From there, we need to figure out a CircleCI strategy - I'll discuss with @andyguest and @selvach .

File: package.json [hunk 2]
```diff
@@ -27,23 +27,27 @@
...
     "test:coverage": "jest --maxWorker=2 --coverage --reporters=default --reporters=jest-junit --ci && coveralls < coverage/lcov.info",
+    "test:e2e:ios": "detox test ./e2e --configuration ios.sim.release",
     "test:i18n": "ts-node ./scripts/i18n-check-locale.ts",
     "test:updateSnapshots": "jest -u",
     "test:watch": "jest --watch",
     "ts-node": "ts-node",
     "web": "expo start --web",
-    "write-icon-names": "node writeIconNames.js"
+    "write-icon-names": "node writeIconNames.js",
+    "build:ios": "react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'"
```
Comment (juliatan) [inline package.json:50]: FWIW, I needed to add an extra config parameter to the end of this line to get it to build properly 🤷🏻‍♀️:
` --assets-dest ios`

Comment (bionara) [reply package.json:50]: Thanks, adding now.

**Review by selvach — APPROVED** [review]

**Pull request merged**